staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -327,8 +328,14 @@ func patchResource(
 			if err := applyPatchToObject(codec, defaulter, currentObjMap, originalPatchMap, versionedObjToUpdate, versionedObj); err != nil {
 				return nil, err
 			}
-			// Convert the object back to unversioned.
-			gvk := kind.GroupKind().WithVersion(runtime.APIVersionInternal)
+
+			var gvk schema.GroupVersionKind
+			if currentGVK.Version == "" {
```

Comment:
I'd expect to either convert to internal or do no conversion, not conversion to some other external version

Reply:
Done.